src/components/Tooltip/hook.tsx

```diff
@@ -0,0 +1,37 @@
+import { TooltipComponent } from "./types";
+import { useContext, MouseEvent } from "react";
+import { TooltipContext } from "./context";
+
+export function useTooltip<T extends HTMLElement>(
+  TooltipComponent: TooltipComponent
+) {
+  const dispatch = useContext(TooltipContext);
+
+  function handleMouseEnter(event: MouseEvent<T>) {
```

Comment:
Probably will need to call `event.persist()` or assign it to a constant, since this event is synthetic. Need further investigation.